app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedController.kt

```diff
@@ -64,23 +65,23 @@ class DefaultRecentlyClosedController(
         }
     }
 
-    override fun handleShare(item: ClosedTab) {
+    override fun handleShare(item: RecoverableTab) {
         navController.navigate(
             RecentlyClosedFragmentDirections.actionGlobalShareFragment(
                 data = arrayOf(ShareData(url = item.url, title = item.title))
             )
         )
     }
 
-    override fun handleRestore(item: ClosedTab) {
-        item.restoreTab(
-            store,
-            sessionManager,
-            onTabRestored = {
-                activity.openToBrowser(
-                    from = BrowserDirection.FromRecentlyClosed
-                )
-            }
+    override fun handleRestore(item: RecoverableTab) {
+        tabsUseCases.restore(item)
+
+        store.dispatch(
+            RecentlyClosedAction.RemoveClosedTabAction(item)
+        )
```

Comment:
Not sure if later we could add a use case for this

Reply:
Yeah, probably, I guess ideally all action dispatching is hidden behind a nicer use case. I guess we could have a specific restore use case that calls into the restore above and then dispatches this additionally.